src/core/scope.c

```diff
@@ -131,23 +131,14 @@ static void scope_set_state(Scope *s, ScopeState state) {
 }
 
 static int scope_add_default_dependencies(Scope *s) {
-        int r;
-
         assert(s);
 
         if (!UNIT(s)->default_dependencies)
                 return 0;
 
-        /* Make sure scopes are unloaded on shutdown */
-        r = unit_add_two_dependencies_by_name(
-                        UNIT(s),
-                        UNIT_BEFORE, UNIT_CONFLICTS,
-                        SPECIAL_SHUTDOWN_TARGET, true,
-                        UNIT_DEPENDENCY_DEFAULT);
-        if (r < 0)
-                return r;
-
-        return 0;
+        /* Make sure scopes are unloaded on shutdown, unless we are meant to survive soft reboot, in which
+         * case we need to conflict with non-soft-reboot targets. */
+        return unit_add_dependencies_on_real_shutdown_targets(UNIT(s));
```

Comment:
I am not convinced this is appropriate.

we already have a knob for turning this off, it's DefaultDependencies=no.

soft-reboot services by definition will run very early and very late at boot, hence they should always set DEfaultDependencies=no anyway

Reply:
> soft-reboot services by definition will run very early and very late at boot

Not really? There's no actual use case I actually have where that is true. Especially the 'early at boot', as in - 'real' boot, it's definitely not the case

Reply:
if system comes back after soft-reboot such services are so early they are already running when we even start doing things.

Reply:
Ordering doesn't affect units that are already running, we already know that, this is no different. On softreboot they are never stopped, so ordering is a no-op, that bit is natural. On 'real' boot on the other hand, the unit is not running, so it needs to start when it's expected to start and respect ordering.